From 092752cac3f1821b2a72a2d255a4538b78ed543d Mon Sep 17 00:00:00 2001
From: Doug Goldstein <cardoe@cardoe.com>
Date: Mon, 17 Feb 2025 17:19:16 -0600
Subject: [PATCH] ironic: fix logging of validation errors

When validation of the node fails, since switching to the SDK the
address of the ValidationResult object is displayed instead of the
actual message. This has been broken since patch
Ibb5b168ee0944463b996e96f033bd3dfb498e304.

Change-Id: I8fbdaadd125ece6a3050b2fbb772a7bd5d7e5304
Signed-off-by: Doug Goldstein <cardoe@cardoe.com>
---
 nova/virt/ironic/driver.py | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/nova/virt/ironic/driver.py b/nova/virt/ironic/driver.py
index 9d6bd32126..7279af09ad 100644
--- a/nova/virt/ironic/driver.py
+++ b/nova/virt/ironic/driver.py
@@ -1213,14 +1213,20 @@ class IronicDriver(virt_driver.ComputeDriver):
         ):
             # something is wrong. undo what we have done
             self._cleanup_deploy(node, instance, network_info)
+            deploy_msg = ("No Error" if validate_chk['deploy'].result
+                          else validate_chk['deploy'].reason)
+            power_msg = ("No Error" if validate_chk['power'].result
+                         else validate_chk['power'].reason)
+            storage_msg = ("No Error" if validate_chk['storage'].result
+                           else validate_chk['storage'].reason)
             raise exception.ValidationError(_(
                 "Ironic node: %(id)s failed to validate. "
                 "(deploy: %(deploy)s, power: %(power)s, "
                 "storage: %(storage)s)")
                 % {'id': node.id,
-                   'deploy': validate_chk['deploy'],
-                   'power': validate_chk['power'],
-                   'storage': validate_chk['storage']})
+                   'deploy': deploy_msg,
+                   'power': power_msg,
+                   'storage': storage_msg})
 
         # Config drive
         configdrive_value = None
-- 
2.39.5 (Apple Git-154)
